### PR TITLE
Update flake8 plugins and clean typos

### PR DESCRIPTION
**Pull request by Goldziher**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

File: tests/plugins/piccolo_orm/piccolo_app.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     Venue,
 )
 
-CURRENT_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+CURRENT_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
```
Comment (jtraub) [inline tests/plugins/piccolo_orm/piccolo_app.py:18]: Maybe we should use here pathlib for consistency?

Comment (jtraub) [reply tests/plugins/piccolo_orm/piccolo_app.py:18]: If we want to use pathlib only then `flake8-use-pathlib` seems to be a good addition to pre-commit hooks

Maybe, `flake8-use-fstring` is also worth checking out.

**Review by cofin — APPROVED** [review]

**Review by peterschutt — APPROVED** [review]

File: starlite/cache/memcached_cache_backend.py [hunk 2]
```diff
@@ -80,8 +80,7 @@ async def set(self, key: str, value: Any, expiration: int) -> None:  # pylint: d
             None
         """
 
-        await self._memcached_client.set(key.encode(), self._config.serialize(value), exptime=expiration)
-        return None
+        await self._memcached_client.set(key.encode(), self._config.serialize(value), expiration=expiration)
```
Comment (seladb) [inline starlite/cache/memcached_cache_backend.py:83]: @Goldziher this change will not work with `aiomcache` - the parameter name is `exptime`:
https://github.com/aio-libs/aiomcache/blob/20f4b18d5fc4c54047ebaf6d28e6024d80d591e4/aiomcache/client.py#L252

File: tests/caching/test_memcached_cache_backend.py [hunk 3]
```diff
@@ -19,9 +19,9 @@ def __init__(self) -> None:
     async def get(self, key: bytes) -> Optional[bytes]:
         return self._cache.get(key)
 
-    async def set(self, key: bytes, value: bytes, exptime: int) -> None:
+    async def set(self, key: bytes, value: bytes, expiration: int) -> None:
```
Comment (seladb) [inline tests/caching/test_memcached_cache_backend.py:22]: Let's change back to `exptime`?

**Review by seladb — CHANGES_REQUESTED** [review]

File: tests/caching/test_memcached_cache_backend.py (hunk 3 above)
Comment (Goldziher) [reply tests/caching/test_memcached_cache_backend.py:22]: Ofc

**Review by infohash — APPROVED** [review]

**Review by seladb — APPROVED** [review]

**Pull request merged**